python/pyspark/sql/streaming/stateful_processor_api_client.py

```diff
@@ -0,0 +1,173 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from enum import Enum
+import os
+import socket
+from typing import Union, cast, Tuple
+
+import pyspark.sql.streaming.StateMessage_pb2 as stateMessage
```

Comment:
Ah okay. We don't protobuf dependencies in plain tests. We shouldn't import this. I think this is the problem.

Reply:
If you need this as a required dependency, we should check if dependencies are existing, and throw a proper error messages.

Reply:
Refer to `pyspark.sql.connect.check_dependencies`

Reply:
Thanks @HyukjinKwon, I'll give it a try!